underfs/cos/src/main/java/alluxio/underfs/cos/COSInputStream.java

```diff
@@ -62,22 +68,42 @@
    * @param key the key of the file
    * @param client the client for COS
    * @param position the position to begin reading from
+   * @param retryPolicy the retry policy to use
    * @param multiRangeChunkSize the chunk size to use on this stream
+   *
    */
   COSInputStream(String bucketName, String key, COSClient client, long position,
-      long multiRangeChunkSize) throws IOException {
+      RetryPolicy retryPolicy, long multiRangeChunkSize) throws IOException {
     super(multiRangeChunkSize);
     mBucketName = bucketName;
     mKey = key;
     mCosClient = client;
     mPos = position;
+    mRetryPolicy = retryPolicy;
     ObjectMetadata meta = mCosClient.getObjectMetadata(mBucketName, key);
     mContentLength = meta == null ? 0 : meta.getContentLength();
   }
 
   @Override
   protected InputStream createStream(long startPos, long endPos)
       throws IOException {
+    // TODO(lu) only retry when object does not exist because of eventual consistency
```

Comment:
why is this a TODO and not addressed here?

Reply:
This is related to many storages that I have no experience using them. Setup the storage account and get the exact exception that those storages will throw when key/object cannot be found may take longer time. That's why I marked as TODO here

Reply:
Addressed the TODOs. We only want to catch and retry Object/bucket not found exception. Thankas